docs/TOKENS.md

```diff
@@ -13,133 +13,141 @@ Spoilers. My love for Astro has grown deeper.
+To create an endpoint in Astro, add a Javascript or Typescript file to the `pages` directory. The name of the file should have the datatype extension. So your file name will look like `<name>.<return-type>.ts`. The Javascript and Typescript extensions will be removed during the build process.
 
-The file path we will be using for generating our token is `pages/rtc/[channel]/[role]/[uid].json.ts`. This endpoint will return JSON data, so the file name ends with `.json.ts`. Our endpoint will also need parameters. In Astro, you use brackets for the file/folder name. Since the Typescript extension gets removed, and the `pages` directory doesn't show up in the URL, our final endpoint will be: `rtc/{channel}/{role}/{uid}.json`.
+The file path we will use to generate our token is `pages/api/token.json.ts`. This endpoint will return JSON data, so the file name ends with `.json.ts`. I like to put all endpoints within an `api` folder so that they are separated from front-end routes. Since the Typescript extension gets removed and the `pages` directory doesn't show up in the URL, our final endpoint will be `api/token.json`.
 
 ## Define the endpoint
-Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
-
-We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
-
+Because our endpoint will need some input information, we will use a `POST` request and retrieve the information from the body. You can define this in Astro by exporting a `POST` function and returning a Response object.
 
 ```ts
 import type { APIContext } from "astro";
 
-export async function GET({ params }: APIContext) {
-  return new Response(
-    JSON.stringify({
-      rtcToken: "token",
-    })
-  )
-}
-```
-
-## CORS
-We first need to set up our headers so that we don't encounter any CORS issues when using our backend.
-
-```ts
-export async function GET({ params }: APIContext) {
-    const headers = new Headers({
-        "Access-Control-Allow-Origin": "*",
-        "Access-Control-Allow-Methods": "GET, OPTIONS",
-    });
+export async function POST({ request }: APIContext) {
+    const { channel, role, uid, expireTime } = await request.json()
 
     return new Response(JSON.stringify({
-        rtcToken: "token"
-    }), { headers })
+        token: "token"
+    }), { status: 200 })
 }
 ```
 
+## Response Helper Functions
+We will abstract away the Response objects in a `utils/sendResponse.ts` file to simplify our endpoint. 
+
+First, we need to set up our headers to avoid any CORS issues when using our backend.
```

Comment:
You should add a transition to make it clear that you are editing the `sendResponse.ts` because moving from the previous sentence to this one it isn't that clear.

Reply:
Updated to:

Within this file, we need to set up our headers to avoid any CORS issues when using our backend.